### PR TITLE
Only validate unfinished cache runs

### PR DESCRIPTION
**Pull request by vdbergh**
Old runs often do not validate due to the existence race conditions in the past.

### CODE REVIEW

File: server/fishtest/rundb.py [hunk 1]
```diff
@@ -87,13 +87,21 @@ def schedule_tasks(self):
         self.scheduler.add_task(180.0, self.validate_random_run)
 
     def validate_random_run(self):
-        run_list = list(self.run_cache.values())
+        # Excess of caution. Another thread may chanche run_cache
```
Comment (vondele) [inline server/fishtest/rundb.py:90]: chanche -> change

**Comment by ppigazzini** [comment]
So far so good (I loaded many old runs in my browser).
```yaml
Jun 01 06:35:13 tests.stockfishchess.org systemd[1]: Stopping Fishtest Server port 6543...
Jun 01 06:35:13 tests.stockfishchess.org pserve[12453]: flush
Jun 01 06:35:14 tests.stockfishchess.org pserve[12453]: ........................done
Jun 01 06:35:16 tests.stockfishchess.org systemd[1]: Stopped Fishtest Server port 6543.
Jun 01 06:35:53 tests.stockfishchess.org systemd[1]: Started Fishtest Server port 6543.
Jun 01 06:35:56 tests.stockfishchess.org pserve[15218]: Starting server in PID 15218.
Jun 01 06:36:00 tests.stockfishchess.org pserve[15218]: Request_task: no useful cached runs left
Jun 01 06:36:00 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:37:01 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:38:05 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:38:47 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665ab125062b2c3cf814ff7c...
Jun 01 06:39:17 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:40:18 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:41:24 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:41:53 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665aad3f062b2c3cf814ff66...
Jun 01 06:42:07 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:43:12 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:44:15 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:44:56 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665aab2d062b2c3cf814ff58...
Jun 01 06:45:15 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:46:15 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:47:16 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:47:48 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665aad3f062b2c3cf814ff66...
Jun 01 06:48:16 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:49:17 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:50:17 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:50:40 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665a9821062b2c3cf814fed8...
Jun 01 06:50:41 tests.stockfishchess.org pserve[15218]: dead task: run: https://tests.stockfishchess.org/tests/view/665a9821062b2c3cf814fed8 task_id: 745 worker: nikkorelox1234-3cores-71adf505-965a
Jun 01 06:51:17 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:52:19 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:53:20 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:53:46 tests.stockfishchess.org pserve[15218]: Validate_random_run: validated aggregated data in cache run 665ab622062b2c3cf814ff94...
Jun 01 06:54:20 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:55:28 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
Jun 01 06:56:33 tests.stockfishchess.org pserve[15218]: Request_task: refresh queue
```

File: server/fishtest/rundb.py (hunk 1 above)
Comment (vdbergh) [reply server/fishtest/rundb.py]: Fixed!

**Review by ppigazzini — APPROVED** [review]
Looks good on PROD

**Pull request merged**